test/1_test_fullpath.js

```diff
@@ -84,14 +87,881 @@ contract('Voucher tests', async (addresses) => {
+      const transactionBlock = await web3.eth.getBlock(transaction.blockNumber);
+      assert.isTrue(
+        voucherStatus.complainPeriodStart.eq(new BN(transactionBlock.timestamp))
+      );
+    });
+
+    it('mark non-redeemed voucher as expired', async () => {
+      const statusBefore = await contractVoucherKernel.vouchersStatus(
+        tokenVoucherKey2
+      );
+
+      // [1000.0000] = hex"80" = 128 = COMMITTED
+      assert.equal(
+        web3.utils.toHex(statusBefore.status),
+        web3.utils.numberToHex(128),
+        'initial voucher status not as expected (COMMITTED)'
+      );
+
+      // fast-forward for a year
+      await timemachine.advanceTimeSeconds(constants.SECONDS_IN_DAY * 365);
+      const expTx = await contractVoucherKernel.triggerExpiration(
+        tokenVoucherKey2
+      );
+
+      truffleAssert.eventEmitted(
+        expTx,
+        'LogExpirationTriggered',
+        (ev) => {
+          return (
+            ev._tokenIdVoucher.eq(tokenVoucherKey2) &&
+            ev._triggeredBy === users.deployer.address
+          );
+        },
+        'expiration not triggered successfully'
+      );
+
+      //Check VoucherKernel state
+      const voucherStatus = await contractVoucherKernel.vouchersStatus(
+        tokenVoucherKey2
+      );
+
+      //[1001.0000] = hex"90" = 144 = EXPIRED
+      assert.equal(
+        web3.utils.toHex(voucherStatus.status),
+        web3.utils.numberToHex(144),
+        'end voucher status not as expected (EXPIRED)'
+      );
+    });
+
+    it('mark voucher as finalized', async () => {
+      await contractBosonRouter.redeem(tokenVoucherKey1, {
+        from: users.buyer.address,
+      });
+
+      //fast forward 8 days (complain period is 7)
+      await timemachine.advanceTimeSeconds(constants.SECONDS_IN_DAY * 8);
+
+      const txFinalize = await contractVoucherKernel.triggerFinalizeVoucher(
+        tokenVoucherKey1,
+        {
+          from: users.buyer.address,
+        }
+      );
+
+      truffleAssert.eventEmitted(
+        txFinalize,
+        'LogFinalizeVoucher',
+        (ev) => {
+          return ev._tokenIdVoucher.eq(tokenVoucherKey1);
+        },
+        'voucher not finalized successfully'
+      );
+
+      //Check VoucherKernel state
+      const voucherStatus = await contractVoucherKernel.vouchersStatus(
+        tokenVoucherKey1
+      );
+      assert.isTrue(voucherStatus.status.eq(new BN(194)));
+    });
+
+    it('must fail: unauthorized redemption', async () => {
+      await truffleAssert.reverts(
+        contractBosonRouter.redeem(tokenVoucherKey1, {
+          from: users.attacker.address,
+        }),
         truffleAssert.ErrorType.REVERT
       );
     });
   });
 
-  describe('Voucher Sets (ERC1155)', () => {
-    it('adding one new order / promise', async () => {
+  //HS:  All other withdraw functions are tested in 3_withdrawals.js. Do we want to move this one?. Withdrawal of deposit not included here
```

Comment:
IMHO moving them into 3_withdrawals.js would be more appropriate

Reply:
OK. Le'ts discuss during the standup if we want to do that now or pick it up when we are refactoring the other file and moving to TypeScript (which I believe is the plan).

Reply:
We forgot to talk about it during the standup. I just talked to Ali about it. Let's agree to move it in  the  3_withdrawals.js story. I'll add it as a comment in the story.